lib/plug/redoc_ui.ex

```diff
@@ -40,18 +40,64 @@ defmodule Redoc.Plug.RedocUI do
       </style>
     </head>
     <body>
-      <redoc spec-url="<%= spec_url %>"></redoc>
+      <redoc
+        <%= for {k, v} <- redoc_opts do %>
+         <%= k %>="<%= v %>"
+        <% end %>
+      ></redoc>
       <script src="https://cdn.jsdelivr.net/npm/redoc@<%= Plug.HTML.html_escape(redoc_version) %>/bundles/redoc.standalone.js"></script>
     </body>
   </html>
   """
 
+  @redoc_options [
+    :disable_search,
+    :download_definition_url,
+    :download_file_name,
+    :expand_default_server_variables,
+    :expand_responses,
+    :expand_single_schema_field,
+    :generated_payload_samples_max_depth,
+    :hide_download_button,
+    :hide_fab,
+    :hide_hostname,
+    :hide_loading,
+    :hide_schema_pattern,
+    :hide_schema_titles,
+    :hide_single_request_sample_tab,
+    :json_sample_expand_level,
+    :lazy_rendering,
+    :max_displayed_enum_values,
+    :menu_toggle,
+    :min_character_length_to_init_search,
+    :native_scrollbars,
+    :nonce,
+    :only_required_in_samples,
+    :path_in_middle_panel,
+    :payload_sample_idx,
+    :required_props_first,
+    :schema_expansion_level,
+    :scroll_y_offset,
+    :show_extensions,
+    :show_object_schema_examples,
+    :show_webhook_ver,
+    :side_nav_style,
+    :simple_one_of_type_label,
+    :sort_enum_values_alphabetically,
+    :sort_operations_alphabetically,
+    :sort_props_alphabetically,
+    :sort_tags_alphabetically,
+    :spec_url,
+    :theme,
+    :untrusted_spec
+  ]
+
   @impl true
   def init(opts) do
-    spec_url = Keyword.fetch!(opts, :spec_url)
```

Comment:
I think this would cause breaking change due to spec_url is missing. We should warn it and merge to redoc_opts if it's set.